features/lorawan/LoRaWANStack.h

```diff
@@ -0,0 +1,290 @@
+     */
+    void commission_device(lora_dev_commission_t commission_data);
+
+    /** Gets commissioning data.
+     *
+     * \return                  The current parameters of the device on success, NULL on failure.
+     */
+    lora_mac_status_t get_commissioning_data(lora_dev_commission_t& commission);
+
+    /** End device OTAA join.
+     *
+     * Based on the LoRaWAN standard 1.0.2.
+     * Join the network using the Over The Air Activation (OTAA) procedure.
+     *
+     * \param  params   The `lorawan_connect_t` type structure.
+     * \return          The state of the LoRa MAC.
+     */
+    lora_mac_status_t join_request_by_otaa(const lorawan_connect_t &params);
+
+    /** End device ABP join.
+     *
+     * Based on the LoRaWAN standard 1.0.2.
+     * Join the network using the Activation By Personalization (ABP) procedure.
+     *
+     * \param  params   The `lorawan_connect_t` type structure.
+     * \return          The state of the LoRa MAC.
+     */
+    lora_mac_status_t activation_by_personalization(const lorawan_connect_t &params);
+
+    /** Sends a message to the gateway.
+     *
+     * @param port              The application port number. Port numbers 0 and 224
+     *                          are reserved, whereas port numbers from 1 to 223
+     *                          (0x01 to 0xDF) are valid port numbers.
+     *                          Anything out of this range is illegal.
+     *
+     * @param data              A pointer to the data being sent. The ownership of the
+     *                          buffer is not transferred. The data is copied to the
+     *                          internal buffers.
+     *
+     * @param length            The size of data in bytes.
+     *
+     * @param flags             A flag used to determine what type of
+     *                          message is being sent, for example:
+
+     *                          MSG_UNCONFIRMED_FLAG = 0x01
+     *                          MSG_CONFIRMED_FLAG = 0x02
+     *                          MSG_MULTICAST_FLAG = 0x04
+     *                          MSG_PROPRIETARY_FLAG = 0x08
+
+     *                          MSG_MULTICAST_FLAG and MSG_PROPRIETARY_FLAG can be
+     *                          used in conjunction with MSG_UNCONFIRMED_FLAG and
+     *                          MSG_CONFIRMED_FLAG depending on the intended use.
+     *
+     *                          MSG_PROPRIETARY_FLAG|MSG_CONFIRMED_FLAG mask will set
+     *                          a confirmed message flag for a proprietary message.
+     *                          MSG_CONFIRMED_FLAG and MSG_UNCONFIRMED_FLAG are
+     *                          mutually exclusive.
+     *                          message is sent, for example:
+     *
+     *
+     * @return                  The number of bytes sent, a negative error code on failure.
+     */
+    int16_t handle_tx(uint8_t port, const uint8_t* data,
```

Comment:
`handle_tx` instead of write ? same for `handle_rx`

the doc above does seem to be trimmed `message is sent, for example:`

Reply:
It's not exactly write(). This initiates a process for TX. If duty cycle is off, it sends immidiately. If duty cycle is on, it only schedules the transmission. In my opinion handle_tx() is better suited name. Comment is right, it returns the number of bytes sent from your message.